plugin/dnstap/msg/msg.go

```diff
@@ -10,100 +9,89 @@ import (
 	"github.com/miekg/dns"
 )
 
-// Builder helps to build Data by being aware of the dnstap plugin configuration.
 type Builder struct {
-	Full bool
-	Data
+	Packed      []byte
+	SocketProto tap.SocketProtocol
+	SocketFam   tap.SocketFamily
+	Address     []byte
+	Port        uint32
+	TimeSec     uint64
```

Comment:
Can TimeSec be a `time.Time`?
or does everything here needs to be in a lower level format for easier packing?

Same question for Address (i.e. why not  an `net.IP`?)

And Port, why not just `int`?

Reply:
Yes it is for the Dnstap struct to take the address of those fields, maybe they shouldn't be exported.